app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -25,18 +25,30 @@ def create_sub_plans
         job_invocation = JobInvocation.find(input[:job_invocation_id])
         proxy_selector = RemoteExecutionProxySelector.new
 
-        # composer creates just "pattern" for template_invocations because target is evaluated
-        # during actual run (here) so we build template invocations from these patterns
-        job_invocation.targeting.hosts.map do |host|
+        ::Host.where(:id => current_batch).map do |host|
```

Comment:
I saw `current_batch` is not defined for `RunHostsJob` when testing this, `include Dynflow::Action::WithBulkSubPlans` helped